src/transformers/models/xglm/modeling_xglm.py

```diff
@@ -338,9 +337,14 @@ def forward(
                     f"Attention mask should be of size {(bsz, 1, tgt_len, src_len)}, but is {attention_mask.size()}"
                 )
             attn_weights = attn_weights.view(bsz, self.num_heads, tgt_len, src_len) + attention_mask
+            attn_weights = torch.max(attn_weights, torch.tensor(torch.finfo(attn_weights.dtype).min))
             attn_weights = attn_weights.view(bsz * self.num_heads, tgt_len, src_len)
 
-        attn_weights = nn.functional.softmax(attn_weights, dim=-1)
+        # upcast to fp32 if the weights are in fp16. Please see https://github.com/huggingface/transformers/pull/17437
+        if attn_weights.dtype == torch.float16:
+            attn_weights = nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32).to(attn_weights.dtype)
```

Comment:
```suggestion
            attn_weights = nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32).to(torch.float16)
```
Maybe this is clearer since some people might think that this line will:
1- First assign `attn_weights = nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32)`, therefore `attn_weights.dtype = torch.float32`
2- Then assign `nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32)` to `torch.float32` 
Even though it is not what will happen maybe just in terms of readability/understandability of the code it is better 🤔 ? Wdyt?

Reply:
I am OK for this change as it is minimal and indeed a bit more clear.